config/sync/migrate_plus.migration.d7_file_media_document.yml

```diff
@@ -0,0 +1,59 @@
+uuid: a2370256-c702-4129-8e0e-72aefb8dcace
+langcode: en
+status: true
+dependencies: {  }
+id: d7_file_media_document
+class: Drupal\migrate\Plugin\Migration
+field_plugin_method: alterFieldInstanceMigration
+cck_plugin_method: null
+migration_tags:
+  - dept_sites
+migration_group: migrate_group_files
+label: 'Migrate D7 files to media document entities'
+source:
+  plugin: file_entity
+  track_changes: true
+  type: document
+  batch_size: 1000
+  constants:
+    bundle: document
+process:
+  bundle: constants/bundle
+  langcode:
+    plugin: default_value
+    source: language
+    default_value: und
+  name:
+    -
+      plugin: get
+      source:
+        - field_file_title
+        - filename
+    -
+      plugin: callback
+      callable: array_filter
+    -
+      plugin: callback
```

Comment:
This odd pipeline enables the following idea: use the title field value but if it's empty for any reason use the filename instead.